processing/src/main/java/org/apache/druid/segment/incremental/IncrementalIndex.java

```diff
@@ -824,24 +797,18 @@ private long getMaxTimeMillis()
 
   public List<String> getDimensionNames()
   {
-    synchronized (dimensionDescs) {
-      return ImmutableList.copyOf(dimensionDescs.keySet());
-    }
+    return ImmutableList.copyOf(dimensionDescs.keySet());
```

Comment:
I tried looking for usages of this method and there don't seem to be any that mutate the list, so I guess it would be safe. That been said, we would probably have to linearly scan/copy the keyset  to transform it to a list, so defensively making it immutable is probably a good practice.

Also, Google [claims](https://github.com/google/guava/wiki/ImmutableCollectionsExplained#copyof-is-smarter-than-you-think) that `copyOf` is smarter than naive copy, although I'm not sure if that's true for copies of mutable objects.